src/server/journal/serializer.cc

```diff
@@ -195,7 +195,11 @@ io::Result<journal::ParsedEntry> JournalReader::ReadEntry() {
   entry.dbid = dbid_;
   entry.opcode = opcode;
 
-  if (opcode == journal::Op::PING || opcode == journal::Op::FIN) {
+  if (opcode == journal::Op::PING) {
+    SET_OR_UNEXPECT(ReadUInt<uint64_t>(), entry.lsn);
```

Comment:
we currently send ping in replica takeover flow.
so you need to read entry.lsn only if master version is ver4 or higher